src/edu/washington/escience/myria/operator/Apply.java

```diff
@@ -47,47 +57,152 @@
   /**
    * @return the {@link #emitEvaluators}
    */
-  public ArrayList<GenericEvaluator> getEmitEvaluators() {
+  public List<GenericEvaluator> getEmitEvaluators() {
     return emitEvaluators;
   }
 
   /**
+   * @param evaluators the evaluators to set
+   */
+  public void setEmitEvaluators(final List<GenericEvaluator> evaluators) {
+    emitEvaluators = ImmutableList.copyOf(evaluators);
+  }
+
+  /**
+   * @return if there are no multivalued emit expressions
+   */
+  private boolean onlySingleValuedExpressions() {
```

Comment:
I understand that this seems inefficient, but it's only called once per `TupleBatch`, and I like to avoid materializing redundant state whenever possible. I don't have a very strong opinion, though.